pkg/detectors/mongodb/mongodb.go

```diff
@@ -46,30 +51,47 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 
 		if verify {
-			func() {
-				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
-				defer cancel()
-				client, err := mongo.Connect(ctx, options.Client().ApplyURI(resMatch))
-				if err != nil {
-					return
-				}
-				defer func() {
-					if err := client.Disconnect(ctx); err != nil {
-						return
-					}
-				}()
-				if err := client.Ping(ctx, readpref.Primary()); err != nil {
-					return
-				}
-				s1.Verified = true
-			}()
+			timeout := s.timeout
+			if timeout == 0 {
+				timeout = defaultTimeout
+			}
+			err := verifyUri(resMatch, timeout)
+			s1.Verified = err == nil
+			if !isErrDeterminate(err) {
+				s1.VerificationError = err
+			}
 		}
 		results = append(results, s1)
 	}
 
 	return results, nil
 }
 
+func isErrDeterminate(err error) bool {
```

Comment:
suggestion: I think you can simplify this with:
```
if e, ok := err.(topology.ConnectionError); ok {
		_, ok = e.Unwrap().(*auth.Error)
		return ok
	}
	return false
```

or i guess if we want to use the errors pkg:
```
func isErrDeterminate(err error) bool {
	var connErr topology.ConnectionError
	if errors.As(err, &connErr) {
		var authErr *auth.Error
		return errors.As(connErr.Unwrap(), &authErr)
	}
	return false
}
```

Reply:
i actually specifically wrote this not to be as terse as possible, but to make it easy to extend with other error cases in the future. do you find it hard to read as it is?

Reply:
ahh okay, i think that makes sense. I sorta had the feeling, but wasn't sure. If we leave the switches might be worth adding the default case to each, no mandatory though. One of the the go linters happens to yell at me when i have a switch with no default. 🤷